pkg/instance/service.go

```diff
@@ -415,22 +499,18 @@ func (s service) DeployDeployment(token string, deployment *model.Deployment) er
 
 		// Add parameters not supplied by user
 		for name, stackParameter := range stack.Parameters {
-			log.Println("name:", name)
-			log.Println("stack parameter:", stackParameter)
-			if _, ok := instanceParameters[name]; !ok && stackParameter.DefaultValue == nil {
-				return fmt.Errorf("required parameter missing: %s", name)
-			}
+			if _, ok := instanceParameters[name]; !ok {
+				instanceParameter := model.DeploymentInstanceParameter{
+					ParameterName: name,
+					StackName:     stack.Name,
+				}
 
-			instanceParameter := model.DeploymentInstanceParameter{
-				ParameterName: name,
-				StackName:     stack.Name,
-			}
+				if stackParameter.DefaultValue != nil {
+					instanceParameter.Value = *stackParameter.DefaultValue
+				}
 
-			if stackParameter.DefaultValue != nil {
-				instanceParameter.Value = *stackParameter.DefaultValue
+				instanceParameters[name] = instanceParameter
```

Comment:
Why do we set the `instanceParameters[name]` if the `stackParameter.DefaultValue == nil`?

And why is this loop done for all and not just `!Consumed` parameters? 🤔